### ISSUE TITLE
Forbid `$t` and friends in `data`

### ISSUE THREAD

**Issue by allanlewis** [intlify/eslint-plugin-vue-i18n#304]
**Please describe what the rule should do:**

Use of `$t` and friends in components' `data` sections should be forbidden because it results in non-dynamic translations.

**What category should the rule belong to?**

- [ ] Enforces code style (layout)
- [x] Warns about a potential error (problem)
- [ ] Suggests an alternate way of doing something (suggestion)
- [ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
data() {
  return {
    foo: this.$t('foo'),
  }
},
```

**Additional context**
